valhalla/baldr/rapidjson_utils.h

```diff
@@ -148,6 +153,74 @@ inline boost::optional<rapidjson::Value&> get_child_optional(V&& v, const char*
   return c;
 }
 
+template <class Ptree> void add_value(const Value& v, Ptree& pt) {
+  switch (v.GetType()) {
+    case kObjectType:
+      add_object(v.GetObject(), pt);
+      break;
+    case kArrayType:
+      add_array(v.GetArray(), pt);
+      break;
+    case kNullType:
+      pt.put("", "null");
+      break;
+    case kFalseType:
+      pt.put("", false);
+      break;
+    case kTrueType:
+      pt.put("", true);
+      break;
+    case kStringType:
+      pt.put("", v.GetString());
+      break;
+    case kNumberType:
+      if (v.IsInt64())
+        pt.put("", v.GetInt64());
+      else if (v.IsUint64())
+        pt.put("", v.GetUint64());
+      else if (v.IsDouble())
+        pt.put("", v.GetDouble());
+      else
+        throw std::runtime_error("unhandled number");
+      break;
+  }
+}
+
+template <class Ptree> void add_object(const GenericObject<true, Value::ValueType>& o, Ptree& pt) {
+  for (const auto& e : o)
+    add_value(e.value, pt.add_child(e.name.GetString(), Ptree{}));
+}
+
+template <class Ptree> void add_array(const GenericArray<true, Value::ValueType>& a, Ptree& pt) {
+  for (const auto& e : a)
+    add_value(e, pt.push_back(std::make_pair("", Ptree{}))->second);
+}
+
+template <class Ptree>
+void read_json(std::basic_istream<typename Ptree::key_type::value_type>& stream, Ptree& pt) {
```

Comment:
this is the important bit. it has the same signature that is found in the `property_tree::json_parser` and so once i implemented it i didnt really have to change anything but the namespace and include in all the things that used it. this is basically the start of the recursion of making a ptree